test/upload.test.js

```diff
@@ -137,13 +137,18 @@ test('Can instantiate Uploader', (t) => {
   t.is(datapub.api, config.api)
 })
 
-test('Push works with packaged dataset', async (t) => {
-  const authzApi = "http://localhost:5000"
-  const token = await client.ckanAuthz(authzApi)
-                            .then(response => response.result.token )
-  await client.push(file, token)
+test('Can get JWT token', async (t) => {
+  const token = await client.doBlobAuthz()
 
   t.is(ckanAuthzMock.isDone(), true)
+  t.is(token, "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzY29wZXMiOiIiLCJ== ")
+})
+
+test('Push works with packaged dataset', async (t) => {
+  const path = 'test/fixtures/sample.csv'
+  const resource = f11s.open(path)
+  await client.pushBlob(resource)
```

Comment:
How we can make sure that `pushBlob()` actually sends the right data in the request? I think in the mock when you get the body in the `.post()` you should validate the `body`

Reply:
maybe after we have a real body working, we can get and insert it in mock body.

Reply:
Sorry what you mean? The `body` basically should be tested with `resource`, to make sure that the `body` converted from `resource` correct. What you mean by saying real body?

Reply:
No idea, I will try to use the last modifications in datapub, and we can create an issue to manage these tests with a real and valid body.

Reply:
See my implementation in the next comment. We can discuss this later in the chat

Reply:
I believe it has nothing to do with datapub, we should test it independent of that in our tests. OK not big deal will create an issue later, but if you check my implementation it should be clear